src/extractor/restriction_parser.cpp

```diff
@@ -160,7 +173,42 @@ RestrictionParser::TryParse(const osmium::Relation &relation) const
             break;
         }
     }
-    return boost::make_optional(std::move(restriction_container));
+
+    // parse conditional tags
+    if (parse_conditionals)
+    {
+        osmium::tags::KeyFilter::iterator fi_begin(filter, tag_list.begin(), tag_list.end());
+        osmium::tags::KeyFilter::iterator fi_end(filter, tag_list.end(), tag_list.end());
+        for (; fi_begin != fi_end; ++fi_begin)
+        {
+            const std::string key(fi_begin->key());
+            const std::string value(fi_begin->value());
+            // Parse condition and add independent value/condition pairs
+            const auto &parsed = osrm::util::ParseConditionalRestrictions(value);
+
+            if (parsed.empty())
+            {
+                osrm::util::Log(logWARNING) << "Conditional restriction parsing failed for \""
+                                            << fi_begin->value() << "\" at the turn "
+                                            << restriction_container.restriction.from.way << " -> "
+                                            << restriction_container.restriction.via.node << " -> "
+                                            << restriction_container.restriction.to.way;
+                continue;
+            }
+
+            for (auto &p : parsed)
+            {
+                std::vector<util::OpeningHours> parsed = util::ParseOpeningHours(p.condition);
+                if (!parsed.empty())
+                    restriction_container.restriction.condition = std::move(parsed);
+            }
+        }
+    }
+
+    // push back a copy of turn restriction
+    parsed_restrictions.push_back(boost::make_optional(restriction_container));
```

Comment:
okay, bug, ticketed boostorg/optional 30

if you want to avoid a copy here you can do

    optional<YourType> tmp{std::move(v)};
    r.push_back(std::move(tmp));